R/utilities-figures.R

```diff
@@ -343,26 +344,44 @@ createPlotsFromExcel <- function(simulatedScenarios, observedData, projectConfig
   })
   names(multiPanelPlots) <- dfPlotGrids$name
 
+  if (nrow(dfExportConfigurations) > 0) {
+    # create a list of ExportConfiguration objects from dfExportConfigurations
+    exportConfiguration <- createEsqlabsExportConfiguration(projectConfiguration)
+    exportConfigurations <- apply(select(dfExportConfigurations, -plotGridName), 1, .createConfigurationFromRow, defaultConfiguration = exportConfiguration)
+    # export plotGrid if defined in exportConfigurations
+    lapply(seq_along(exportConfigurations), function(i) {
+      exportConfigurations[[i]]$savePlot(multiPanelPlots[[dfExportConfigurations$plotGridName[i]]])
+    })
+  }
+
   return(multiPanelPlots)
 }
 
 #' @keywords internal
-.plotConfigurationFromRow <- function(...) {
-  plotConfiguration <- createEsqlabsPlotConfiguration()
-  lapply(seq_along(...), function(i) {
-    value <- ...[[i]]
-    colName <- names(...)[[i]]
+.createConfigurationFromRow <- function(defaultConfiguration, ...) {
+  columns <- c(...)
+  newConfiguration <- defaultConfiguration$clone()
+
+  lapply(seq_along(columns), function(i) {
+    value <- columns[[i]]
+    colName <- names(columns)[[i]]
     if (!is.na(value)) {
-      # numeric columns
-      if (colName %in% c("xAxisLimits", "yAxisLimits")) {
-        plotConfiguration[[colName]] <- eval(parse(text = value))
+      # Check if the field name is supported by the configuration class
+      if (!.validateClassHasField(object = newConfiguration, field = colName)){
```

Comment:
@hannaei I added a check if the property that the user tries to set is supported by the configuration.